### PR TITLE
added storageClass to CRD

### PR DESCRIPTION
**Pull request by bitscuit**
Needed to generalize `mergeChangedMap` function to handle values which were not resource values, and handle cases where user explicitly sets value to null.

### CODE REVIEW

File: controllers/operandconfig.go [hunk 1]
```diff
@@ -141,8 +141,18 @@ func mergeChangedMap(key string, defaultMap interface{}, changedMap interface{},
 			//Check if the value was set, otherwise set it
 			if changedMap == nil {
 				finalMap[key] = defaultMap
+			} else if defaultMap == nil {
+				// need to handle case where user explicitly set value to null
+				finalMap[key] = changedMap
```
Comment (horis233) [inline controllers/operandconfig.go:146]: @bitscuit Could you explain why this is required?

From my understanding, finalMap[key] is always equal to changedMap.

Comment (bitscuit) [reply controllers/operandconfig.go:146]: good catch. Will remove it. It was there initially to solve an issue, but the comparable key check solves it as well.

**Comment by bitscuit** [comment]
Added missing elastic-stack-operator configuration.

File: controllers/operandconfig.go [hunk 2]
```diff
@@ -142,7 +142,14 @@ func mergeChangedMap(key string, defaultMap interface{}, changedMap interface{},
 			if changedMap == nil {
 				finalMap[key] = defaultMap
 			} else {
-				finalMap[key], _ = rules.ResourceComparison(defaultMap, changedMap)
+				comparableKeys := []string{"replicas", "cpu", "memory"}
+				for _, comparable := range comparableKeys {
```
Comment (bitscuit) [inline controllers/operandconfig.go:146]: not sure if we want to do the comparable key check like this. If we add more configurations in the future, and squads determine they can be changed after deployment, then comparing this way will not scale well.

**Comment by horis233** [comment]
> Added missing elastic-stack-operator configuration.

elastic-stack-operator has been deprecated, we don't need to add it.

File: controllers/operandconfig.go (hunk 2 above)
Comment (horis233) [reply controllers/operandconfig.go:146]: I think we can add the comparable key check since the ResourceComparison is for resource size comparison.

Comment (bitscuit) [reply controllers/operandconfig.go:146]: right, but I mean in the future, if we add any more configurations, e.g. maybe operators include a new `healthcheck-interval` configuration that is the number of seconds between pinging a health check endpoint, and we want to add this to CS CR as well, then this new variable could be comparable (maybe we want to take the lowest value).

The way this code is checking for comparable keys won't scale very well if we add more configurations like this in the future. We would also need to keep updating the list every time. Should we consider this? Or is looping over a list of keys good enough?

Comment (horis233) [reply controllers/operandconfig.go:146]: I think we keep them for now. The price to maintain the list is not high for us. 
Considering the case you list, we need to update the ResourceConparison function anyway without the comparable keys.

**Review by Daniel-Fan — APPROVED** [review]
@horis233 
I have OK with the PR. Could you please review it again?

**Review by horis233 — APPROVED** [review]
/lgtm

**Pull request merged**